src/game_targets.cpp

```diff
@@ -16,91 +16,70 @@
  */
 
 // Headers
-#include <vector>
-#include "main_data.h"
 #include "game_targets.h"
-
-static std::vector<RPG::SaveTarget>& data = Main_Data::game_data.targets;
-
-namespace Game_Targets {
-	std::vector<RPG::SaveTarget>::iterator FindTarget(int id, bool create);
-}
-
-std::vector<RPG::SaveTarget>::iterator Game_Targets::FindTarget(int id, bool create) {
-	std::vector<RPG::SaveTarget>::iterator it;
-	for (it = data.begin(); it != data.end(); ++it)
-		if (it->ID == id)
-			return it;
-	if (!create)
-		return data.end();
-	data.resize(data.size() + 1);
-	data.back().ID = id;
-	return data.end() - 1;
+#include <algorithm>
+
+template <typename T>
+static auto FindTarget(T&& targets, int map_id) {
+	return std::lower_bound(targets.begin(),
```

Comment:
Considering that games usually use very few targets is a binary search worth it?

Reply:
Changed to linear search